### PR TITLE
Updated analyzers to VS2017.

### PR DESCRIPTION
**Pull request by azchohfi**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Analyzers now build and work on VS2017.

### :arrow_heading_down: What is the current behavior?
Only builds on VS2015.

### :new: What is the new behavior (if this is a feature change)?
ONLY builds on VS2017.

### :boom: Does this PR introduce a breaking change?
Yes, build only regarding build environments. Analyzers works just as they should.

### :bug: Recommendations for testing
The only problem with this PR is that the test project is now referencing Mono.Android, Java.Interop and Xamarin.iOS assemblies with a **full path**.
$(MSBuildProgramFiles32) does not work anymore on VS2017.
If anybody know how to make the tests pass without referencing the reference assemblies with a full path 

### :memo: Links to relevant issues/docs
N/A

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Nuspec files were updated (when applicable)
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by martijn00** [comment]
What about something like: `<Import Project="$(MSBuildExtensionsPath)\Xamarin\iOS\Xamarin.iOS.CSharp.targets" />`

We had something similar in:
https://github.com/appveyor/ci/issues/1448

**Comment by Cheesebaron** [comment]
Not sure what exactly you are asking about here @martijn00

**Comment by martijn00** [comment]
It might be a clue to solve why there is a hardcoded path right now. That variable could be used i think.

File: CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj [hunk 1]
```diff
@@ -118,29 +146,9 @@
...
-    </Reference>
-    <Reference Include="System.Composition.Runtime, Version=1.0.30.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.30\lib\portable-net45+win8+wp8+wpa81\System.Composition.Runtime.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="System.Composition.TypedParts, Version=1.0.30.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.30\lib\portable-net45+win8+wp8+wpa81\System.Composition.TypedParts.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Xamarin.iOS">
+      <HintPath>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\Xamarin.iOS.dll</HintPath>
```
Comment (martijn00) [inline CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj]: A problem with this is that you hardcode `Enterprise`, while i for example us the `community` edition. I think we really need to solve this another way.

Comment (azchohfi) [inline CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj]: I agree, but I don't know how...

Comment (martijn00) [inline CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj]: @onovotny Do you have any idea about this one?

Comment (onovotny) [inline CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj]: You should not be referencing with a full path....

Comment (onovotny) [inline CodeAnalysis/MvvmCross.CodeAnalysis.Test/MvvmCross.CodeAnalysis.Test.csproj]: You don't need the hintpath at all

**Pull request merged**